fe/fe-core/src/main/java/org/apache/doris/statistics/util/StatisticsUtil.java

```diff
@@ -974,4 +976,28 @@ public static boolean isMvColumn(TableIf table, String columnName) {
             || columnName.startsWith(CreateMaterializedViewStmt.MATERIALIZED_VIEW_AGGREGATE_NAME_PREFIX);
     }
 
+    public static boolean isEmptyTable(TableIf table, AnalysisInfo.AnalysisMethod method) {
+        int waitRowCountReportedTime = 90;
+        if (!(table instanceof OlapTable) || method.equals(AnalysisInfo.AnalysisMethod.FULL)) {
+            return false;
+        }
+        OlapTable olapTable = (OlapTable) table;
+        for (int i = 0; i < waitRowCountReportedTime; i++) {
+            if (olapTable.getRowCount() > 0) {
+                return false;
+            }
+            // If visible version is 2, table is probably not empty. So we wait row count to be reported.
```

Comment:
Where does this assumption come from and why it is helpful to wait?

Reply:
Table visible version == 2 usually means use loaded the first batch of data to the table, in this case, the table is probably not empty. In this case, we wait the row count to be reported so that we can avoid get all 0 statistics. If visible version is 1 (init), the table is empty for sure. If it is > 2, we don't know it's empty or not, because of delete or truncate and so on. In this case we don't wait, simply use the row count.